src/db/Procedures.ts

```diff
@@ -1,99 +1,37 @@
-import log from 'electron-log';
+import { ConnectionModel } from './Models';
+import { store } from './redux/store';
 
-const { Client } = require('pg');
-
-type DBProcedure = {
-  routine_catalog: string;
-  routine_name: string;
-};
-
-type DBQuery = {
-  datname: string;
-};
+const model = new ConnectionModel();
+model.nickname = 'something_dumb';
+model.username = 'kpmg';
+model.password = 'asdf';
+model.address = 'localhost';
 
 export default class Procedures {
-  address: string;
-
-  port: number;
-
   public async getProceduresForDB(
     databases: string[]
   ): Promise<Map<string, string[]>> {
     const res = new Map<string, string[]>();
     await Promise.all(
       databases.map(async (database) => {
-        const procedures = await this.fetchProcedures(database);
+        const procedures = await this.fetchProcedures();
         res.set(database, procedures);
       })
     );
     return res;
   }
 
   public async getDatabases() {
-    const database = 'React';
-    const client = await new Client({
-      host: this.address,
-      port: this.port,
-      password: 'asdf',
-      user: 'kpmg',
-      database,
-    });
-    client.connect();
-    const result = await client.query(
-      `select datname from pg_catalog.pg_database where datistemplate = false`
-    );
-    client.end();
-    return Promise.all(
-      result.rows.map((row: DBQuery) => {
-        return row.datname;
-      })
-    );
+    return store.getState().connection.serverConnection.getDatabasesQuery();
```

Comment:
To cleanup the code a bit you could just put
```ts
store.getState().connection.serverConnection
```
 in the constructor and assign it to a serverConnection property for the class

Reply:
wait but then when the serverConnection is updated the property will need to be updated? Wouldn't that get rid of the whole point of the global

Reply:
That's why I'm accessing it directly so that if we select a new ServerConnection we can still just call the methods, I don't think the methods should be tied to instance of Procedures, otherwise what's the point of the global in the first place.